### PR TITLE
add-interview-request-api-and-send-interview

### PR DESCRIPTION
**Pull request by SMSaeedi**

### CODE REVIEW

File: src/main/kotlin/com/linchpino/core/controller/InterviewController.kt [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+class InterviewController(private val service: InterviewService) {
+
+    @Operation(summary = "Create a new Interview")
+    @ResponseStatus(HttpStatus.CREATED)
+    @ApiResponses(
+        value = [ApiResponse(responseCode = "201", description = "Interview added successfully"),
+            ApiResponse(responseCode = "400", description = "Invalid request body")]
+    )
+    @PostMapping(consumes = [MediaType.APPLICATION_JSON_VALUE], produces = [MediaType.APPLICATION_JSON_VALUE])
+    fun newInterview(@Valid @RequestBody request: InterviewRequest): ResponseEntity<InterviewResult> {
```
Comment (m-tilab) [inline src/main/kotlin/com/linchpino/core/controller/InterviewController.kt]: Please rename following method and  classes:
newInterview -> createInterview
CreateInterviewRequest -> InterviewRequest 
InterviewResult -> CreateInterviewResult

File: src/main/kotlin/com/linchpino/core/dto/CreateAccountRequest.kt [hunk 2]
```diff
@@ -21,4 +21,4 @@ data class CreateAccountResult(
 	val email: String,
 	val type: AccountTypeEnum = AccountTypeEnum.UNKNOWN,
 	val status: AccountStatus = AccountStatus.DEACTIVATED,
-)
+)
```
Comment (m-tilab) [inline src/main/kotlin/com/linchpino/core/dto/CreateAccountRequest.kt]: discard this change

File: src/main/kotlin/com/linchpino/core/dto/InterviewRequest.kt [hunk 3]
```diff
@@ -0,0 +1,18 @@
+package com.linchpino.core.dto
+
+import jakarta.validation.constraints.NotNull
+
+data class InterviewRequest(
```
Comment (m-tilab) [inline src/main/kotlin/com/linchpino/core/dto/InterviewRequest.kt]: Rename to CreateInterviewRequest

File: src/main/kotlin/com/linchpino/core/dto/InterviewRequest.kt [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+    @field:NotNull(message = "jobPositionId is required") val jobPositionId: Long,
+    @field:NotNull(message = "interviewTypeId is required") val interviewTypeId: Long,
+    @field:NotNull(message = "timeSlotId is required") val timeSlotId: Long,
+    @field:NotNull(message = "jobSeekerEmail is required") val jobSeekerEmail: String,
+)
+
+data class InterviewResult(
```
Comment (m-tilab) [inline src/main/kotlin/com/linchpino/core/dto/InterviewRequest.kt]: Rename to CreateInterviewResult

File: src/main/kotlin/com/linchpino/core/service/AccountService.kt [hunk 5]
```diff
@@ -13,13 +13,18 @@ import org.springframework.transaction.annotation.Transactional
...
-	fun createAccount(createAccountRequest: CreateAccountRequest): CreateAccountResult {
-		val account: Account = mapper.accountDtoToAccount(createAccountRequest)
-		// encrypt password
-		account.password = passwordEncoder.encode(account.password)
-		repository.save(account)
-		return mapper.entityToResultDto(account)
-	}
+    fun createAccount(createAccountRequest: CreateAccountRequest): CreateAccountResult {
+        val account: Account = mapper.accountDtoToAccount(createAccountRequest)
+        //account.createdOn(LocalDate.now())
```
Comment (m-tilab) [inline src/main/kotlin/com/linchpino/core/service/AccountService.kt:27]: Why is this line commented on? We need to have the createOn date on table

File: src/main/kotlin/com/linchpino/core/service/InterviewService.kt [hunk 6]
```diff
@@ -0,0 +1,41 @@
...
+@Transactional
+class InterviewService(
+    private val repository: InterviewRepository,
+    private val accountRepository: AccountRepository,
+) {
+
+    fun newInterview(request: InterviewRequest): InterviewResult {
+        val email = request.jobSeekerEmail
+        val interview: Interview = InterviewMapper.interviewDtoToInterview(request)
+        //interview.createdOn(LocalDate.now())
```
Comment (m-tilab) [inline src/main/kotlin/com/linchpino/core/service/InterviewService.kt]: Also, interview.createdOn needed

**Review by m-tilab — COMMENTED** [review]
When I started the project, I got the following error and the application stopped:
Caused by: org.springframework.data.mapping.PropertyReferenceException: No property 'isUserExistByEmail' found for type 'Account'

File: src/main/kotlin/com/linchpino/core/controller/InterviewController.kt (hunk 1 above)
Comment (SMSaeedi) [reply src/main/kotlin/com/linchpino/core/controller/InterviewController.kt]: done

File: src/main/kotlin/com/linchpino/core/service/InterviewService.kt (hunk 6 above)
Comment (SMSaeedi) [reply src/main/kotlin/com/linchpino/core/service/InterviewService.kt]: removed

File: src/main/kotlin/com/linchpino/core/service/AccountService.kt (hunk 5 above)
Comment (SMSaeedi) [reply src/main/kotlin/com/linchpino/core/service/AccountService.kt:27]: removed

File: src/main/kotlin/com/linchpino/core/dto/InterviewRequest.kt (hunk 4 above)
Comment (SMSaeedi) [reply src/main/kotlin/com/linchpino/core/dto/InterviewRequest.kt]: done

File: src/main/kotlin/com/linchpino/core/dto/InterviewRequest.kt (hunk 3 above)
Comment (SMSaeedi) [reply src/main/kotlin/com/linchpino/core/dto/InterviewRequest.kt]: done

File: src/main/kotlin/com/linchpino/core/dto/CreateAccountRequest.kt (hunk 2 above)
Comment (SMSaeedi) [reply src/main/kotlin/com/linchpino/core/dto/CreateAccountRequest.kt]: this class has been removed.

File: pom.xml [hunk 7]
```diff
@@ -17,13 +17,12 @@
...
-        <postgres.version>42.7.2</postgres.version>
-        <mapstruct.version>1.5.5.Final</mapstruct.version>
-        <lombok.version>1.18.24</lombok.version>
-    </properties>
+	<properties>
+		<kotlin.version>1.9.22</kotlin.version>
+		<openAPI.version>2.3.0</openAPI.version>
+		<postgres.version>42.7.2</postgres.version>
+		<mapstruct.version>1.5.5.Final</mapstruct.version>
+	</properties>
```
Comment (m-tilab) [inline pom.xml]: We have an unchanged code difference here. Please review the changes first. You can compare your branch with the develop branch and apply only the necessary changes. These changes were raised when we had formatting on all sources

File: src/test/kotlin/com/linchpino/core/service/interview/InterviewServiceTest.kt [hunk 8]
```diff
@@ -0,0 +1,114 @@
...
+
+        // Then
+        assertEquals(createInterviewResult, result)
+        verify(interviewRepo, times(1)).save(captor.capture())
+        val savedInterview = captor.value
+        assertEquals("john.doe@example.com", savedInterview.jobSeekerAccount?.email)
+        assertEquals("Mentor.Mentoriii@example.com", savedInterview.mentorAccount?.email)
+        assertEquals(AccountStatusEnum.DEACTIVATED, savedInterview.jobSeekerAccount?.status)
+    }
+}
```
Comment (m-tilab) [inline src/test/kotlin/com/linchpino/core/service/interview/InterviewServiceTest.kt:184]: Please same as IT, write a gray scenario

File: src/main/kotlin/com/linchpino/core/dto/CreateInterviewRequest.kt [hunk 9]
```diff
@@ -0,0 +1,21 @@
...
+data class CreateInterviewRequest(
+    @field:NotNull(message = "jobPositionId is required") val jobPositionId: Long,
+    @field:NotNull(message = "interviewTypeId is required") val interviewTypeId: Long,
+    @field:NotNull(message = "timeSlotId is required") val timeSlotId: Long,
+    @field:NotNull(message = "mentorAccId is required") val mentorAccId: Long,
+    @field:Email(message = "jobSeekerEmail is required") val jobSeekerEmail: String,
+)
+
+data class CreateInterviewResult(
+    val id: Long?,
```
Comment (m-tilab) [inline src/main/kotlin/com/linchpino/core/dto/CreateInterviewRequest.kt]: rename to interviewId

File: src/main/kotlin/com/linchpino/core/dto/CreateInterviewRequest.kt [hunk 10]
```diff
@@ -0,0 +1,21 @@
...
+    val jobPositionId: Long?,
+    val interviewTypeId: Long?,
+    val timeSlotId: Long?,
+    val mentorAccId: Long?,
```
Comment (m-tilab) [inline src/main/kotlin/com/linchpino/core/dto/CreateInterviewRequest.kt]: mentorAccountId

File: pom.xml (hunk 7 above)
Comment (SMSaeedi) [reply pom.xml]: done, the thing on the below lines was I removed some break lines between the dependencies.

File: src/main/kotlin/com/linchpino/core/dto/CreateInterviewRequest.kt (hunk 9 above)
Comment (SMSaeedi) [reply src/main/kotlin/com/linchpino/core/dto/CreateInterviewRequest.kt]: done

File: src/main/kotlin/com/linchpino/core/dto/CreateInterviewRequest.kt (hunk 10 above)
Comment (SMSaeedi) [reply src/main/kotlin/com/linchpino/core/dto/CreateInterviewRequest.kt]: done

File: src/main/kotlin/com/linchpino/core/dto/CreateInterviewRequest.kt [hunk 11]
```diff
@@ -0,0 +1,21 @@
+package com.linchpino.core.dto
+
+import jakarta.validation.constraints.Email
+import jakarta.validation.constraints.NotNull
+
+data class CreateInterviewRequest(
```
Comment (m-tilab) [inline src/main/kotlin/com/linchpino/core/dto/CreateInterviewRequest.kt:6]: image
As you see in the above image, the class name can be "CreateInterviewDTO"

**Comment by SMSaeedi** [comment]
added new PR instead

**Pull request closed without merging**